### PR TITLE
Spark: Remove reflection workarounds from v3.1 code

### PR DESCRIPTION
**Pull request by wypoon**
This is a follow-up to #3415. In this change, we remove reflection workarounds from the v3.1 code.

### CODE REVIEW

File: spark/v3.0/build.gradle [hunk 1]
```diff
@@ -129,7 +129,6 @@ project(":iceberg-spark:iceberg-spark3-extensions") {
     compileOnly project(':iceberg-data')
     compileOnly project(':iceberg-orc')
     compileOnly project(':iceberg-common')
-    compileOnly project(':iceberg-spark')
```
Comment (wypoon) [inline spark/v3.0/build.gradle:132]: This and the ones below are not necessary. I removed them so they are not consistent with the build.gradle for v.31 and v3.2.

File: spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala [hunk 2]
```diff
@@ -100,7 +99,7 @@ trait AssignmentAlignmentSupport {
             case StructType(fields) =>
               // build field expressions
               val fieldExprs = fields.zipWithIndex.map { case (field, ordinal) =>
-                createAlias(GetStructField(col, ordinal, Some(field.name)), field.name)
+                Alias(GetStructField(col, ordinal, Some(field.name)), field.name)()
```
Comment (rdblue) [inline spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala:102]: I think that we should leave the reflection in 3.0. We aren't renaming the `iceberg-spark3-runtime` Jar so it is possible that people currently using it for 3.1 will continue to. As long as we can maintain compatibility with that by doing nothing, I think we should. I don't think that it is necessary to do much else, but I don't see a reason to remove the compatibility.

File: spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/sql/TestDeleteFrom.java [hunk 3]
```diff
@@ -41,70 +38,6 @@ public void removeTables() {
     sql("DROP TABLE IF EXISTS %s", tableName);
   }
 
-  @Test
-  public void testDeleteFromUnpartitionedTable() {
-    // This test fails in Spark 3.1. `canDeleteWhere` was added to `SupportsDelete` in Spark 3.1,
-    // but logic to rewrite the query if `canDeleteWhere` returns false was left to be implemented
-    // later.
-    Assume.assumeTrue(Spark3VersionUtil.isSpark30());
```
Comment (rdblue) [inline spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/sql/TestDeleteFrom.java:49]: I think Anton added new tests in place of these in 3.2. Can you check whether they should be added here?

**Comment by rdblue** [comment]
@wypoon, this mostly looks good to me. Could you remove the changes for Spark 3.0? I don't see a compelling reason to remove the reflection support and potentially break people using the current runtime module that upgrade to 0.13.0.

File: spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala (hunk 2 above)
Comment (wypoon) [reply spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala]: Ok, I reverted the code changes for spark/v3.0. I kept the change to spark/v3.0/build.gradle as that is just cleanup.

File: spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/sql/TestDeleteFrom.java (hunk 3 above)
Comment (wypoon) [reply spark/v3.1/spark/src/test/java/org/apache/iceberg/spark/sql/TestDeleteFrom.java:49]: I added the new tests from v3.2 to the v3.1 `TestDeleteFrom` but the new tests fail. I'll let @aokolnychyi add correct 3.1 versions if applicable in a separate change.

File: spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala (hunk 2 above)
Comment (rdblue) [reply spark/v3.0/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/AssignmentAlignmentSupport.scala]: Thanks!

File: spark/v3.0/build.gradle (hunk 1 above)
Comment (rdblue) [reply spark/v3.0/build.gradle:132]: Sounds good to me.

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @wypoon!

**Pull request merged**